torch/csrc/jit/passes/onnx/fixup_onnx_controlflow.cpp

```diff
@@ -211,8 +212,10 @@ void FixupONNXLoopNodeInputs(Node* node) {
 
   // add cast to condition input outside the loop.
   Value* cond_val = node->input(1);
-  if (IsCondCastRequired(cond_val))
-    InsertCastForCond(cond_val, graph, node);
+  if (IsCondCastRequired(cond_val)) {
+    auto* cast_node = InsertCastForCond(cond_val, graph, node);
```

Comment:
Can we copyMetadata inside the InsertCastForCond?

Reply:
This cannot be added inside since there are two use cases, one for initial cond outside of loop, and one for cond inside loop. The node to copyMetadata from is different for the two cases.